### PR TITLE
Fix: Only use exact roundtrip mappings from UCM files, skip fallback mappings

### PR DESCRIPTION
**Pull request by tobwen**
This fixes a critical bug where ASCII characters (`0x21-0x7E`) were incorrectly converted to fullwidth Unicode equivalents when reading files with various encodings, particularly `Windows-1251`.

### Problem issued
https://github.com/duckdb/duckdb/issues/19267

### Affected by the bug
- CSV parsing failures (delimiters not recognized)
- Multiple columns merged into one
- Numbers, commas, semicolons displayed as fullwidth (`１，；`  instead of `1,;`)

### Root cause
The converter script was processing ALL mappings from UCM files, including fallback mappings (`|1`, `|2`, `|3`). Since dictionaries store only one value per key, fallback mappings overwrote exact mappings (`|0`), resulting in incorrect character conversions.

### Example
```
  <U0021> \x21 |0  # Correct: EXCLAMATION MARK
  <UFF01> \x21 |1  # Fallback: FULLWIDTH EXCLAMATION MARK (was winning)
```

### Solution
Modified scripts/converter.py to only process exact roundtrip mappings (`|0`), ignoring all fallback mappings (`|1, |2, |3`).
Fixes [19267](https://github.com/duckdb/duckdb/issues/19267).

### CODE REVIEW

**Comment by pdet** [comment]
Hi @tobwen thanks for the contribution!

Can you run the script and the formatter, and manually add the test that triggers the problem you found?

Cheers,

**Comment by pdet** [comment]
> > Can you run the script and the formatter, and manually add the test that triggers the problem you found?
> 
> Of course, I have already rebuilt the whole thing and it has passed all the tests. I can now offer to create a new test that addresses the issue of [duckdb/duckdb#19267](https://github.com/duckdb/duckdb/issues/19267)
> 
> But maybe we need to create better tests overall to include such edge cases?

Yes, a handcrafted test that checks for edge cases will be nice! IIRC the auto-generated tests actually only checks one character per row, so we could also extend that to maybe generate multi-column, with multiple characters per column, string values.

**Comment by daviewales** [comment]
I'm guessing this would also fix https://github.com/duckdb/duckdb-encodings/issues/10.

**Comment by alex-stewart-14** [comment]
Any updates on this getting merged? This is blocking my ability to use DuckDB in production

File: scripts/converter.py [hunk 1]
```diff
@@ -71,10 +71,16 @@ def parse_ucm_to_utf8_map(ucm_path):
...
-			match = re.match(r'<U([0-9A-Fa-f]+)>\s+((?:\\x[0-9A-Fa-f]{2})+)\s+\|\d+', line)
+			match = re.match(r'<U([0-9A-Fa-f]+)>\s+((?:\\x[0-9A-Fa-f]{2})+)\s+\|(\d+)', line)
 			if not match:
 				continue
 
+			# Only process exact roundtrip mappings (fallback indicator = 0)
+			# Skip fallback mappings (|1, |2, |3) to avoid overwriting correct mappings with fullwidth/fallback characters
+			fallback_indicator = int(match.group(3))
+			if fallback_indicator != 0:
+				continue
```
Comment (daviewales) [inline scripts/converter.py:82]: If we're skipping every match where the `fallback_indicator != 0`, why not just update the regex to only match those lines in the first place?


```suggestion
                        # Only process exact roundtrip mappings (fallback indicator = 0)
			# Skip fallback mappings (|1, |2, |3) to avoid overwriting correct mappings with fullwidth/fallback characters
			match = re.match(r'<U([0-9A-Fa-f]+)>\s+((?:\\x[0-9A-Fa-f]{2})+)\s+\|0', line)
            if not match:
                continue
```

**Comment by pdet** [comment]
Should be fixed in https://github.com/duckdb/duckdb-encodings/pull/20
